packages/atomic/src/components/common/atomic-focus-trap.tsx

```diff
@@ -118,14 +126,36 @@ export class AtomicFocusTrap {
 
   @Listen('focusin', {target: 'document'})
   onFocusChanged(e: FocusEvent) {
+    const elementIsPartOfHost = (focusedElement: Element | ShadowRoot) =>
+      isAncestorOf(this.host, focusedElement);
+
+    const elementIsPartOfScope = (focusedElement: Element | ShadowRoot) =>
+      isAncestorOf(this.scope, focusedElement);
+
+    const previousFocusedElementWasFirstOfHost = () =>
+      e.relatedTarget === getFirstFocusableDescendant(this.host);
+
     if (!e.target || !this.active) {
       return;
     }
 
     const focusedElement = getFocusedElement();
 
-    if (focusedElement && isAncestorOf(this.host, focusedElement)) {
-      return;
+    if (focusedElement) {
+      if (
+        elementIsPartOfHost(focusedElement) ||
+        !elementIsPartOfScope(focusedElement)
+      ) {
+        return;
+      }
+      if (
+        previousFocusedElementWasFirstOfHost() &&
+        !elementIsPartOfHost(focusedElement) &&
+        elementIsPartOfScope(focusedElement)
+      ) {
+        getLastFocusableDescendant(this.host)?.focus;
+        return;
+      }
     }
 
     getFirstFocusableDescendant(this.host)?.focus();
```

Comment:
I think there's a small issue here. When the focus ceases to exist (which is usually when it leaves the page), then we focus back into the modal, even when the scope isn't the entire page.

Basically, I think we want the following interactions when moving the focus from one place to another where:
* If `Page` and `Scope` are the same, then we don't disregard any interactions with `Page`.
* `Scope` excludes `Trap`
* The start of a line represents the relatedTarget, and the end of a line represents the current focus.

```mermaid
sequenceDiagram
  participant Page
  participant Scope
  participant Trap
  Page->>Page: Do nothing (out of scope).
  Page->>Scope: Focus first element (out of trap).
  Page->>Trap: Do nothing (already in trap).
  Scope->>Page: Do nothing (out of scope).
  Scope->>Scope: Focus first element (out of trap).
  Scope->>Trap: Do nothing (already in trap).
  Trap->>Page: Do nothing (out of scope)
  Trap->>Scope: Focus first **or** last element (out of trap)
  Trap->>Trap: Do nothing (already in trap).
```

Or, reordered by reasons:
```mermaid
sequenceDiagram
  participant Page
  participant Scope
  participant Trap

  Note over Page,Trap: Current focus is outside the scope
  Page->>Page: Do nothing (out of scope).
  Scope->>Page: Do nothing (out of scope).
  Trap->>Page: Do nothing (out of scope)

  Note over Page,Trap: Current focus is already inside the trap
  Page->>Trap: Do nothing (already in trap).
  Scope->>Trap: Do nothing (already in trap).
  Trap->>Trap: Do nothing (already in trap).

  Note over Page,Trap: Current focus is in a bad place
  Page->>Scope: Focus first element (out of trap).
  Scope->>Scope: Focus first element (out of trap).
  Trap->>Scope: Focus first **or** last element (out of trap)
```

All in all, it looks like we could use an algorithm like this:
```ts
const focusedElement = getFocusedElement();
const previousFocusedElement = e.relatedTarget ?? document.body;
const isInTrap = focusedElement && elementIsPartOfHost(focusedElement);
const isInScope = focusedElement && elementIsPartOfScope(focusedElement);
const wasInTrap = elementIsPartOfHost(previousFocusedElement);
const wasInScope = elementIsPartOfScope(previousFocusedElement);

if (!isInScope) {
  return;
}
if (isInTrap) {
  return;
}
if (!wasInTrap) {
  focusFirstElement();
} else {
  if (previousFocusedElement === getFirstFocusableDescendant()) {
    focusLastElement();
  } else {
    focusFirstElement();
  }
}
```

Reply:
🤔 
When the focus ceases to exist we don't get a `focusin` event, so we don't have to handle that situation.
When the focus comes back into the page, the previous focus is not lost, so we continue the navigation as is. 

Is that what you mean? Or is there another way I don't see to lose the focus

Reply:
Discussed out of PR and we determined the current state was good